### PR TITLE
kcidb: add build output artifacts/info

### PR DESCRIPTION
**Pull request by khilman**

### CODE REVIEW

**Comment by khilman** [comment]
Backend output from my local test instance
```{
  "version": "1", 
  "builds": [
    {
      "origin": "kernelci", 
      "description": "v5.6-rc6-35-g3a415c6da16b", 
      "log_url": "http://khilman.ddns.net:8082/amlogic/gitlab-integ/v5.6-rc6-35-g3a415c6da16b/arm64/defconfig/gcc-8/build.log", 
      "start_time": "2020-03-18T23:44:58.136000", 
      "misc": {
        "build_platform": [
          "Linux", 
          "paris", 
          "5.4.0-0.bpo.3-amd64", 
          "#1 SMP Debian 5.4.13-1~bpo10+1 (2020-02-07)", 
          "x86_64", 
          ""
        ], 
        "vmlinux_bss_size": 469928, 
        "build_time": 13.62, 
        "kernel_image_size": 27249152, 
        "vmlinux_data_size": 1932128
      }, 
      "revision_origin_id": "kernelci.org:v5.6-rc6-35-g3a415c6da16b", 
      "duration": 13.62, 
      "compiler": "aarch64-linux-gnu-gcc (Debian 8.3.0-2) 8.3.0", 
      "config_name": "defconfig", 
      "config_url": "http://khilman.ddns.net:8082/amlogic/gitlab-integ/v5.6-rc6-35-g3a415c6da16b/arm64/defconfig/gcc-8/kernel.config", 
      "origin_id": "kernelci.org:5e72b27a46ddc19726d40857", 
      "valid": true, 
      "architecture": "arm64", 
      "revision_origin": "kernelci", 
      "output_files": [
        {
          "url": "http://khilman.ddns.net:8082/amlogic/gitlab-integ/v5.6-rc6-35-g3a415c6da16b/arm64/defconfig/gcc-8/System.map", 
          "name": "System.map"
        }, 
        {
          "url": "http://khilman.ddns.net:8082/amlogic/gitlab-integ/v5.6-rc6-35-g3a415c6da16b/arm64/defconfig/gcc-8/Image", 
          "name": "kernel_image"
        }, 
        {
          "url": "http://khilman.ddns.net:8082/amlogic/gitlab-integ/v5.6-rc6-35-g3a415c6da16b/arm64/defconfig/gcc-8/modules.tar.xz", 
          "name": "modules.tar.xz"
        }
      ]
    }
  ], 
  "revisions": [
    {
      "origin": "kernelci", 
      "git_repository_commit_hash": "3a415c6da16b2e3446a5b46847668e7379412e70", 
      "origin_id": "kernelci.org:v5.6-rc6-35-g3a415c6da16b", 
      "git_repository_branch": "gitlab-integ", 
      "discovery_time": "2020-03-18T23:44:58.136000", 
      "git_repository_commit_name": "v5.6-rc6-35-g3a415c6da16b", 
      "valid": true, 
      "git_repository_url": "https://git.kernel.org/pub/scm/linux/kernel/git/khilman/linux-amlogic.git"
    }
  ]
}
```

File: app/utils/kcidb.py [hunk 1]
```diff
@@ -122,6 +122,19 @@ def push_build(build_id, first, bq_options, db_options={}, db=None):
         })
         bq_data['revisions'] = [bq_revision]
 
+    files = {
+        'kernel_image': build[models.KERNEL_IMAGE_KEY],
+        'modules.tar.xz': build[models.MODULES_KEY],
```
Comment (gctucker) [inline app/utils/kcidb.py]: I think the key should be `modules` here, and the value will be `modules.tar.xz` typically but maybe something else or potentially nothing.

File: app/utils/kcidb.py [hunk 2]
```diff
@@ -122,6 +122,19 @@ def push_build(build_id, first, bq_options, db_options={}, db=None):
...
+        'System.map': build[models.SYSTEM_MAP_KEY],
+    }
+    output_files = []
+    for name, file in files.iteritems():
+        url = "/".join([
+            STORAGE_URL,
+            build[models.FILE_SERVER_RESOURCE_KEY],
+            file])
```
Comment (gctucker) [inline app/utils/kcidb.py]: Exception seen on staging:
```
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]: [   ERROR/MainThread] Task kcidb-build[ade04b95-7fcc-468a-9cfb-e07ba5cacab7] raised unexpected: TypeError('sequence item 2: expected string or Unicode, NoneType found',)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]: Traceback (most recent call last):
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/.venv/api.staging.kernelci.org/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     R = retval = fun(*args, **kwargs)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/.venv/api.staging.kernelci.org/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     return self.run(*args, **kwargs)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/api.staging.kernelci.org/app/taskqueue/tasks/kcidb.py", line 31, in push_build
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     taskc.app.conf.db_options)
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:   File "/srv/api.staging.kernelci.org/app/utils/kcidb.py", line 135, in push_build
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]:     file])
Mar 19 17:02:27 kernelci-staging kernelci-celery[9570]: TypeError: sequence item 2: expected string or Unicode, NoneType found
```
So probably because one file isn't in a build.  I suppose that can easily happen with builds that fail, but I haven't debugged it to see what the data actually was.

**Comment by khilman** [comment]
Yup, problem with null filenames when build fails.  Updated to handle null fields, and tested with a failing build locally.

File: app/utils/kcidb.py (hunk 1 above)
Comment (khilman) [reply app/utils/kcidb.py]: Good point.  Updated changing the key to `modules`

File: app/utils/kcidb.py (hunk 2 above)
Comment (khilman) [reply app/utils/kcidb.py]: You were right.  On failed builds, the `file` variable will be `None` causing this to fail.  Added a check to catch that.

File: app/utils/kcidb.py [hunk 3]
```diff
@@ -122,6 +122,20 @@ def push_build(build_id, first, bq_options, db_options={}, db=None):
...
+        'System.map': build[models.SYSTEM_MAP_KEY],
+    }
+    output_files = []
+    for name, file in files.iteritems():
+        if file:
+            url = "/".join([
+                STORAGE_URL,
+                build[models.FILE_SERVER_RESOURCE_KEY],
+                file])
+            output_files.append({"name": name, "url": url})
```
Comment (gctucker) [inline app/utils/kcidb.py]: Python tip, this could have been written a bit better using a list comprehension:
```python
base_url = urlparse.urljoin(
    STORAGE_URL, build[models.FILE_SERVER_RESOURCE_KEY])
output_files = list(
    {'name': name, 'url': '/'.join([base_url, file])}
    for name, file in files.iteritems() if file
)
```

File: app/utils/kcidb.py [hunk 4]
```diff
@@ -144,8 +158,13 @@ def push_build(build_id, first, bq_options, db_options={}, db=None):
...
             build[models.KERNEL_CONFIG_KEY],
         ]),
+        'output_files': output_files,
         'misc': {
-            'defconfig': build[models.DEFCONFIG_FULL_KEY],
+            'kernel_image_size': build[models.KERNEL_IMAGE_SIZE_KEY],
+            'vmlinux_bss_size': build[models.VMLINUX_BSS_SIZE_KEY],
+            'vmlinux_data_size': build[models.VMLINUX_DATA_SIZE_KEY],
+            'build_platform': build[models.BUILD_PLATFORM_KEY],
+            'build_time': build[models.BUILD_TIME_KEY],
```
Comment (gctucker) [inline app/utils/kcidb.py]: There's already a generic `duration` field for this: https://github.com/kernelci/kcidb/blob/master/kcidb/db/schema.py#L124
which is in fact populated a bit earlier in this function.

So while I think the other parts are less trivial to add to a common schema, I think `build_time` should be dropped from `misc`.

Comment (khilman) [reply app/utils/kcidb.py]: Agreed, forgot about the existing `duration`.  Will remove.

File: app/utils/kcidb.py (hunk 3 above)
Comment (khilman) [reply app/utils/kcidb.py:137]: Yeah, sorry.  I'm not a python expert.  I copied this from the build `output_files` handling.  They should probably both be updated in a follow-up cleanup patch if desired.

Comment (gctucker) [reply app/utils/kcidb.py:137]: Alright I'll do that in a follow-up.

File: app/taskqueue/celeryconfig.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
 
 """Celery configuration values."""
 
-BROKER_URL = "redis://localhost/0"
+BROKER_URL = "redis://redis/0"
```
Comment (gctucker) [inline app/taskqueue/celeryconfig.py:21]: Err, this doesn't seem related to the kcidb change.

Comment (khilman) [reply app/taskqueue/celeryconfig.py]: oops, that's unrelated.  Removed.
